packages/@react-spectrum/s2/src/TableView.tsx

```diff
@@ -194,6 +194,9 @@ export class S2TableLayout<T> extends TableLayout<T> {
 
   protected buildCollection(): LayoutNode[] {
     let [header, body] = super.buildCollection();
+    if (!header) {
```

Comment:
@LFDanLu can you explain when this happens?

Reply:
If I remember correctly, now that we are rendering the contents of the virtualized content even when we haven't fully processed the collection (aka the changes in Virtualizer and rect size check changes) there is a moment where  we haven't processed the collection and thus can't perform build the header/body yet